modules/core/src/main/java/org/apache/ignite/internal/processors/platform/client/ClientRequestHandler.java

```diff
@@ -115,7 +116,7 @@
         int status = e instanceof IgniteClientException ?
             ((IgniteClientException)e).statusCode() : ClientStatus.FAILED;
 
-        return new ClientResponse(req.requestId(), status, e.getMessage());
+        return new ClientResponse(req.requestId(), status, X.getFullStackTrace(e));
```

Comment:
We do not include full stack traces into thin client responses. Thin client users are supposed to check server logs (if they have access) to understand the issue when error is returned.

Reply:
disagree here, previous message (without my changes) has zero info for error detection, check it plz!

Reply:
Then the message should be fixed. We absolutely can not pass server stack traces to the clients.

Reply:
do we have conversions or something else about thin client return messages ?

Reply:
frequently users have no access to server logs, my fix helps to understand problem correctly.

Reply:
Please see this discussion: http://apache-ignite-developers.2346864.n4.nabble.com/Exception-handling-in-thin-client-should-we-pass-stack-traces-to-the-client-td22392.html

If we want to expose server stack traces to thin clients, this should be configurable and disabled by default.